src/lib/Builder.hs

```diff
@@ -152,6 +153,11 @@ emitAtomToName _ (Var v) = return v
 emitAtomToName hint x = emitHinted hint (Atom x)
 {-# INLINE emitAtomToName #-}
 
+emitAtomToNameAnn :: (Builder r m, Emits n) => NameHint -> LetAnn -> Atom r n -> m n (AtomName r n)
+emitAtomToNameAnn _ _ (Var v) = return v
```

Comment:
I think it should be valid to put this special case in `EmitDecl` directly (perhaps with the additional check that `LetAnn` is `PlainLet`. That might catch even more redundant renames. Did you try that?

Reply:
Good idea.  Done.